site/source/tutorials/internationalization.md

```diff
@@ -0,0 +1,504 @@
+
+* `formatUnit`: formats a number for a specific unit, and delegates to [`Globalize.formatUnit`](https://github.com/globalizejs/globalize/blob/master/doc/api/unit/unit-formatter.md).
+* `getUnitFormatter`: creates a factory that formats a number for a specific unit, and delegates to [`Globalize.unitFormatter`](https://github.com/globalizejs/globalize/blob/master/doc/api/unit/unit-formatter.md).
+
+The list of possible units can be found in the [CLDR](https://github.com/unicode-cldr/cldr-units-full/blob/master/main/en/units.json).
+
+```typescript
+// Assuming all relevant CLDR data have been loaded
+import { formatUnit } from '@dojo/i18n/unit';
+
+formatUnit(1, 'speed-mile-per-hour'); // '1 mile per hour'
+formatUnit(65, 'speed-mile-per-hour'); // '65 miles per hour'
+
+formatUnit(1, 'foot', { form: 'long' }); // '1 foot'
+formatUnit(5280, 'foot', { form: 'long' }); // '5,280 feet'
+formatUnit(1, 'foot', { form: 'short' }); // '1 ft'
+formatUnit(5280, 'foot', { form: 'short' }); // '5,280 ft'
+formatUnit(1, 'foot', { form: 'narrow' }); // '1′'
+formatUnit(5280, 'foot', { form: 'narrow' }); // '5,280′'
+```
+
+## Internationalization with `@dojo/widget-core`
+
+While `@dojo/i18n` is designed to be an independent package, nearly every Dojo 2 application will have a view component. So [`@dojo/widget-core`](https://github.com/widget-core/) provides a custom mixin (`@dojo/widget-core/mixins/I18n`) to make working with `@dojo/i18n` more friendly. Widgets that have incorporated this mixin can localize message bundles by passing them to the `localizeBundle` method during rendering. If messages for the widget's locale have not been loaded yet, then the default messages are returned, and the widget is invalidated once the locale-specific messages have loaded. The object returned by the `localizeBundle` method contains all the messages, as well as a format method that takes a message key as well as any options. If message formatting is not required, then messages can be directly accessed (e.g., `messages.hello`).
+
+In addition to the `localizeBundle` method, `@dojo/widget-core/mixins/I18n` introduces two properties: a boolean `rtl` property and a string `locale` property. The boolean `rtl` property determines whether the text decoration is right-to-left (`true`) or left-to-right(`false`: the default). The `locale` property specifies the locale used to determine which message translations are rendered. Since each widget can have its own distinct locale, it is possible to use multiple languages in the same application. If no `locale` property is included, widgets assume the root application locale (`i18n.locale`). Note, however, that since Dojo 2 widgets are controlled, child widgets do not automatically inherit the locale from their parent. Parents must pass the correct locale as needed to child widgets; otherwise, a child widget will use the application locale while its parent uses a custom locale.
```

Comment:
"Dojo 2 widgets are controlled" we probably need to link to what controlled means (perhaps link to the widget tutorial)?